### DISCUSSION TITLE
Is there a way to serialize `Peer` object?

### DISCUSSION THREAD

**Discussion by DeepBhat** [peers/peerjs#1093] [Q&A] [answered]
I wanted to share the `Peer` instance between multiple tabs using a `SharedWorker`, but I realized `Peer` objects are not really serializable and cannot be converted to a string. What are my options? Is it even possible?

**Comment by jonasgloning** [comment, marked as answer]
Serializing is not possible, as you can transfer neither the WebRTC connection nor the connection to the PeerServer.

I think the main problem is that WebRTC itself is not available in a Worker.
Working around that, you could:
- Use PeerJS in one tab and relay messages to all tabs
- Use one PeerJS instance per tab, sharing the IDs through the worker
